### PR TITLE
test: switch options menu list tests to react testing library

### PR DESCRIPTION
**Pull request by lsprr**
Issue #1159

This PR switches the Options Menu List tests to React Testing Library.

Validation steps

- Run yarn test
- See that there are coverage percentages in the table
- Verify that the tests are converted properly

### CODE REVIEW

**Comment by lsprr** [comment]
Hi @scurker. 

I wanted to inform you that I discovered the `closeOnSelect` prop isn't being used in the component's logic. I'm unsure why it's included if it's not utilized. Furthermore, I strongly suggest either removing it entirely or incorporating it into the component's functionality. That's why I haven't tested `closeOnSelect` in the tests.

File: packages/react/src/components/OptionsMenu/OptionsMenuList.test.tsx [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+      <li>option 2</li>
+    </OptionsMenuList>
+  );
+
+  fireEvent.click(screen.getByRole('link'));
+
+  expect(onClick).toBeCalledTimes(1);
+});
+
+test('should focus child links with keydown events', () => {
```
Comment (scurker) [inline packages/react/src/components/OptionsMenu/OptionsMenuList.test.tsx:249]: This test name mentions focus, but it looks like we're checking for `click` here. Is this test right?

**Review by scurker — COMMENTED** [review]
One test stood out to me, so just wanted to check to make sure it's testing the right thing.

**Review by scurker — APPROVED** [review]
🚨 Reviewed for Security 🚨

**Pull request merged**